core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -161,52 +174,37 @@ void *CudaSpace::allocate(const char *arg_label, const size_t arg_alloc_size,
 }
 
 namespace {
-void *impl_allocate_common(const Cuda &exec_space, const char *arg_label,
-                           const size_t arg_alloc_size,
+void *impl_allocate_common(const int device_id,
+                           [[maybe_unused]] const cudaStream_t stream,
+                           const char *arg_label, const size_t arg_alloc_size,
                            const size_t arg_logical_size,
                            const Kokkos::Tools::SpaceHandle arg_handle,
-                           bool exec_space_provided) {
+                           [[maybe_unused]] bool exec_space_provided) {
```

Comment:
should we fine a new name for this in particular since we are not passing an exec space anymore? Maybe: `do_device_sync` (which would have the opposite meaning of the current one) or `stream_sync_only` or something like that?

Reply:
Sure, see a25a2f0.